### PR TITLE
Fix possible IllegalArgumentException if CP has non-file: scheme (#1213)

### PR DESCRIPTION
**Pull request by vorburger**
Signed-off-by: Michael Vorburger <mike@vorburger.ch>

### CODE REVIEW

**Comment by vorburger** [comment]
see https://github.com/gradle/gradle/issues/1213

File: subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
 public class ClasspathUtil {
     public static void addUrl(URLClassLoader classLoader, Iterable<URL> classpathElements) {
         try {
-            Set<URI> original = new HashSet<URI>();
+            Set<URI> original = new HashSet<>();
```
Comment (eriwen) [inline subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java]: We cannot omit type information here as the base-services module needs to be source compatible with Java 6

File: subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java [hunk 2]
```diff
@@ -54,13 +54,17 @@ public static void addUrl(URLClassLoader classLoader, Iterable<URL> classpathEle
     }
 
     public static ClassPath getClasspath(ClassLoader classLoader) {
-        final List<File> implementationClassPath = new ArrayList<File>();
+        final List<File> implementationClassPath = new ArrayList<>();
```
Comment (eriwen) [inline subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java]: Same as above

**Comment by eriwen** [comment]
@vorburger I've provided a review. Thanks for the developer certificate notice. I believe we still need you to [sign a CLA](http://gradle.org/contributor-license-agreement/) please as we aren't set up for developer certificates yet.

File: subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java (hunk 1 above)
Comment (vorburger) [inline subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java]: sorry that actually was my Eclipse Java 8 auto Save Action, not intentional proposed changed; removed now

**Comment by vorburger** [comment]
Re. "test to verify would be appropriate (perhaps an integration test using your sample from #1213)" I suspect I would have to add a dependency to org.apache.felix.framework to repro. it - would that be acceptable? In [subprojects/base-services/base-services.gradle](https://github.com/gradle/gradle/blob/master/subprojects/base-services/base-services.gradle), really OK? Thought it was worth to double check this... ;-) as I suspect that is NOK. PS: Also you guys seem to be using some Groovy BDD test framework thing (Spock?), is that a hard must have requirement, or are plain simple Java JUnit acceptable? If yes, put in new src/test/java, or keep src/test/groovy?

**Comment by oehme** [comment]
>  I suspect I would have to add a dependency to org.apache.felix.framework to repro.

Not to Gradle, just to the project under test.

> Also you guys seem to be using some Groovy BDD test framework thing (Spock?), is that a hard must have requirement

You'll find that the tests are much easier to write using Spock, just take some inspiration from other integration tests.

The test should:

- create a project
- add a dependency on Felix to that project
- add a dependency on the TAPI to that project
- add a simple test class to the project which uses the TAPI
- run `gradle test` on the project and assert that it succeeds

**Comment by bmuschko** [comment]
@vorburger Are you still interested in providing the test? This PR has become kind of stale.

**Comment by bmuschko** [comment]
Thanks again for providing the pull request. We are sorry but we won't be able to merge the pull request without a test. We can reopen the issue in case you are still interested and address the review comments.

**Pull request closed without merging**